url-join/index.d.ts

```diff
@@ -0,0 +1,16 @@
+// Type definitions for url-join v0.8.3
+// Project: https://github.com/jfromaniello/url-join
+// Definitions by: Rogier Schouten <https://github.com/rogierschouten>
+// Definitions: https://github.com/borisyankov/DefinitelyTyped
+
+declare module "url-join" {
```

Comment:
I should add that there is nothing wrong with your declaration per se. the difference here is when using `declare module "url-join"` you are declaring a **global** module called `url-join`, this means that there can only be one version of  it at a time, and if two components depend on two versions of `url-join` an error will be shown. 
if however, you remove `declare module` part, it stays within its scope, and thus multiple versions of this declaration can co-exist as they do not step on each other in the global scope.
Thanks for the numerous contributions!